chart/templates/controller.yaml

```diff
@@ -52,52 +51,59 @@ rules:
 subjects:
   - kind: ServiceAccount
-    name: transport-controller-sa
-    namespace: '{{.Values.wds_cp_name}}-system'
+    name: transport-controller
+    namespace: {{.Release.Namespace}}
 ---
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: transport-controller-config
-  namespace: {{.Values.wds_cp_name}}-system
 data:
-  get-kubeconfig.sh: |
-    #!/bin/bash
-    # this script receives a ControlPlane name and a parameter
-    # that determines whether to extract the ControlPlane's in-cluster kubeconfig
-    # or the external kubeconfig (if set to "true", the first will be retrieved).
-    # The function returns the requested kubeconfig's content in base64.
-    # it assumes the kubeconfig context is set to access the hosting cluster.
-
-    cpname="$1"
-    get_incluster_key="$2"
-
-    key=""
-    if [[ "$get_incluster_key" == "true" ]]; then
-      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.inClusterKey}');
-    else
-      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.key}');
+  get-its-config.sh: |
+    #!/bin/env bash
+    its_name="${1%"-system"}" # ITS name or ITS namespace
+    if [ "$its_name" = "" ] ; then
+        for cp in `kubectl get controlplane -o name`; do
+            cp=${cp##*/}
+            if kubectl get controlplane $cp -o=jsonpath='{.metadata.labels}' | grep "imbs" &> /dev/null ; then
+                if [ "$its_name" = "" ] ; then
+                    its_name=$cp
+                else
+                    >&2 echo "ERROR: found more than one Control Plane of type imbs!"
+                    exit 1
+                fi
+            fi
+        done
+        if [ "$its_name" = "" ] ; then
+            >&2 echo "ERROR: no Control Plane of type imbs found!"
+            exit 2
+        fi
     fi
-
-    # get secret details
-    secret_name=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.name}')
-    secret_namespace=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.namespace}')
-
+    key=$(kubectl get controlplane $its_name -o=jsonpath='{.status.secretRef.inClusterKey}')
+    secret_name=$(kubectl get controlplane $its_name -o=jsonpath='{.status.secretRef.name}')
+    secret_namespace=$(kubectl get controlplane $its_name -o=jsonpath='{.status.secretRef.namespace}')
+    # get the kubeconfig in base64
+    kubectl get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}"
+  get-wds-config.sh: |
+    #!/bin/env bash
+    wds_name="${1%"-system"}" # WDS name or WDS namespace
+    key=$(kubectl get controlplane $wds_name -o=jsonpath='{.status.secretRef.inClusterKey}')
+    secret_name=$(kubectl get controlplane $wds_name -o=jsonpath='{.status.secretRef.name}')
+    secret_namespace=$(kubectl get controlplane $wds_name -o=jsonpath='{.status.secretRef.namespace}')
     # get the kubeconfig in base64
     kubectl get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}"

```

Comment:
why do we split this into two functions that are doing almost the same?

Reply:
@nirrozenbaum I agree that there is some code in common and I can combine them into a single script if you prefer.
The ITS script implements additional functionality to match the behavior required in the PCH.
I thought to keep them separated for clarity.
Let me know if you prefer to combine them.

Reply:
is it possible to implement the extra logic of getting the ITS cp name in a different place and then in this script keep using it as it was before?

Reply:
I unified the 2 scripts... testing before pushing commit